qiskit/algorithms/minimum_eigensolvers/vqe.py

```diff
@@ -0,0 +1,316 @@
+
+        self.estimator = estimator
+        self.ansatz = ansatz
+        self.optimizer = optimizer
+        self.gradient = gradient
+
+        # TODO change VariationalAlgorithm interface to use a public attribute
+        # this has to go via getters and setters due to the VariationalAlgorithm interface
+        self._initial_point = initial_point
+
+    @property
+    def initial_point(self) -> Sequence[float] | None:
+        """Return the initial point."""
+        return self._initial_point
+
+    @initial_point.setter
+    def initial_point(self, value: Sequence[float] | None) -> None:
+        """Set the initial point."""
+        self._initial_point = value
+
+    def compute_minimum_eigenvalue(
+        self,
+        operator: BaseOperator | PauliSumOp,
+        aux_operators: ListOrDict[BaseOperator | PauliSumOp] | None = None,
+    ) -> VQEResult:
+        ansatz = self._check_operator_ansatz(operator)
+
+        initial_point = self.initial_point
+        if initial_point is None:
+            initial_point = np.random.random(ansatz.num_parameters)
+        elif len(initial_point) != ansatz.num_parameters:
+            raise ValueError(
+                f"The dimension of the initial point ({len(self.initial_point)}) does not match "
+                f"the number of parameters in the circuit ({ansatz.num_parameters})."
+            )
+
+        start_time = time()
+
+        eval_energy = self.get_eval_energy(ansatz, operator)
+
+        if self.gradient is not None:
+            eval_gradient = self.get_eval_gradient(ansatz, operator)
+        else:
+            eval_gradient = None
+
+        # perform optimization
+        if callable(self.optimizer):
+            opt_result = self.optimizer(fun=eval_energy, x0=initial_point, jac=eval_gradient)
+        else:
+            opt_result = self.optimizer.minimize(
+                fun=eval_energy, x0=initial_point, jac=eval_gradient
+            )
+
+        eval_time = time() - start_time
+
+        optimal_point = opt_result.x
+        logger.info(
+            f"Optimization complete in {eval_time} seconds.\nFound opt_params {optimal_point}."
+        )
+
+        aux_values = None
+        if aux_operators:
+            # not None and not empty list
+            # TODO this is going to be replaced by eval_observables (see PR #8683)
+            aux_values = self._eval_aux_ops(ansatz, optimal_point, aux_operators)
+
+        result = VQEResult()
+        result.eigenvalue = opt_result.fun + 0j
+        result.aux_operator_eigenvalues = aux_values
+        result.cost_function_evals = opt_result.nfev
+        result.optimal_point = optimal_point
+        result.optimal_parameters = dict(zip(self.ansatz.parameters, optimal_point))
+        result.optimal_value = opt_result.fun
+        result.optimizer_time = eval_time
+        return result
+
+    @classmethod
+    def supports_aux_operators(cls) -> bool:
+        return True
+
+    def get_eval_energy(
```

Comment:
It would be good if these have consistent names with the SamplingVQE 🙂 Personally I would have a slight preference for spelling words out, as we do that in most other places, but as long as it's consistent I'm good! 🙂

Reply:
I went with private methods `_get_evaluate_energy` and `_get_evaluate_gradient`, which return callables `evaluate_energy` and `evaluate_gradient`. I'm happy to rename them if these are unpopular.